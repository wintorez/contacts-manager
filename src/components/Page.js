import React from 'react';
import PropTypes from 'prop-types';
import { Header, Menu, Container, Loader, Message } from 'semantic-ui-react';
import { Link } from 'react-router-dom';
import { Helmet } from 'react-helmet';

const Page = ({
  title,
  location: { pathname },
  loading,
  error,
  hideMenu,
  children
}) => (
  <Container style={{ marginTop: '1rem' }}>
    {title && (
      <Helmet>
        <title>{title} - Contacts Manager Demo</title>
      </Helmet>
    )}
    <Header as="h1" style={{ margin: `1.5rem 0` }}>
      Contacts Manager Demo
    </Header>
    {!hideMenu && (
      <Menu tabular>
        <Menu.Item active={pathname.startsWith('/employees')}>
          <Link to="/employees">Employees</Link>
        </Menu.Item>
        <Menu.Item active={pathname.startsWith('/settings')}>
          <Link to="/settings">Settings</Link>
        </Menu.Item>
      </Menu>
    )}
    {loading && <Loader active inline="centered" />}
    {error && (
      <Message negative>
        Oops! Something went wrong! Please try again later.
      </Message>
    )}
    {!loading && !error && <Container>{children}</Container>}
  </Container>
);

Page.propTypes = {
  title: PropTypes.string,
  location: PropTypes.object.isRequired,
  loading: PropTypes.bool,
  error: PropTypes.bool,
  hideMenu: PropTypes.bool,
  children: PropTypes.node
};

export default Page;
