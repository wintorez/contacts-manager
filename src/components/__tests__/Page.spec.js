import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Page from '../Page';

const defaults = {
  history: {},
  location: { pathname: '/employees' }
};

describe('<Page />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Page {...defaults}>Page Content</Page>
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without content', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Page {...defaults} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when loading', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Page {...defaults} loading>
            Page Content
          </Page>
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with error', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Page {...defaults} error>
            Page Content
          </Page>
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when hiding menu', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Page {...defaults} hideMenu>
            Page Content
          </Page>
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
