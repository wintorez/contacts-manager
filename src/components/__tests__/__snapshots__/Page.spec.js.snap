// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> renders correctly 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui container"
  >
    Page Content
  </div>
</div>
`;

exports[`<Page /> renders correctly when hiding menu 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui container"
  >
    Page Content
  </div>
</div>
`;

exports[`<Page /> renders correctly when loading 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui active centered inline loader"
  />
</div>
`;

exports[`<Page /> renders correctly with error 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui negative message"
  >
    Oops! Something went wrong! Please try again later.
  </div>
</div>
`;

exports[`<Page /> renders correctly without content 1`] = `
<div
  className="ui container"
  style={
    Object {
      "marginTop": "1rem",
    }
  }
>
  <h1
    className="ui header"
    style={
      Object {
        "margin": "1.5rem 0",
      }
    }
  >
    Contacts Manager Demo
  </h1>
  <div
    className="ui tabular menu"
  >
    <div
      className="active item"
      onClick={[Function]}
    >
      <a
        href="/employees"
        onClick={[Function]}
      >
        Employees
      </a>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <a
        href="/settings"
        onClick={[Function]}
      >
        Settings
      </a>
    </div>
  </div>
  <div
    className="ui container"
  />
</div>
`;
